File: PS/Programmers/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include<map>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    map<int, int> mp;
    int n = numbers.size();
    for (int i = 0; i < n - 1; i++)
        for (int j = i + 1; j < n; j++)
        {
            int temp = numbers[i] + numbers[j];
            if (mp.find(temp) == mp.end()) {
                mp[temp] = 1;
                answer.push_back(temp);
            }
        }
    sort(answer.begin(), answer.end());
    return answer;
}

File: PS/Programmers/두 개 뽑아서 더하기.py
def solution(numbers):
    n = len(numbers)
    answer = { }
    answer = set()
    for i in range(n):
        for j in range(i+1,n):
            answer.add(numbers[i] + numbers[j])
    ans = list(answer)
    ans.sort()
    return ans